beetsplug/lastgenre/__init__.py

```diff
@@ -322,26 +322,23 @@ def fetch_track_genre(self, obj):
 
     # Main processing: _get_genre() and helpers.
 
-    def _format_and_stringify(self, tags: list[str]) -> str:
-        """Format to title_case if configured and return as delimited string."""
+    def _format_genres(self, tags: list[str]) -> list[str]:
+        """Format to title_case if configured and return as list."""
         if self.config["title_case"]:
-            formatted = [tag.title() for tag in tags]
+            return [tag.title() for tag in tags]
         else:
-            formatted = tags
-
-        return self.config["separator"].as_str().join(formatted)
+            return tags
 
     def _get_existing_genres(self, obj: LibModel) -> list[str]:
         """Return a list of genres for this Item or Album. Empty string genres
         are removed."""
-        separator = self.config["separator"].get()
```

Comment:
`genre` values in `lastgenre` users libraries are joined using `separator` configuration: we need to take it into account in the migration logic.

Additionally, we need to deprecate this configuration:
1. Add a note in the changelog explaining that it's now only used for migrating their existing genres and that it can be removed once that's done
2. Add a `.. deprecated` admonition in the docs, explaining the above
3. Remove it from `lastgenre` plugin